solvers/NonlinearProgram.m

```diff
@@ -322,6 +322,22 @@
       cnstr_id = obj.next_nlcon_id;
       obj.next_nlcon_id = obj.next_nlcon_id-3;
       obj.nlcon_id = [obj.nlcon_id cnstr_id];
+      
+      if(strcmpi(obj.solver,'studentsnopt'))
+        if(~checkDependency('studentSnopt'))
```

Comment:
This check seems redundant. The obj.solver can only be set through calling `setSolver` function, and we have already checked dependency inside `setSolver`